utils/net_interface_filter.go

```diff
@@ -0,0 +1,99 @@
+package utils
+
+import (
+	"errors"
+	"net"
+	"strings"
+)
+
+// Compares characters until the end of lString is reached or a difference was found.
+func rStrBeginsWithLstring(lString string, rString string, err *error) bool {
+	if len(rString) < len(lString) {
+		(*err) = errors.New("source is bigger than target string")
+		return false
+	}
+
+	ls := strings.ToLower(lString)
+	lt := strings.ToLower(rString)
+
+	for i := 0; i < len(lString); i++ {
+		if ls[i] != lt[i] {
+			return false
+		}
+	}
+
+	return true
+}
+
+// Filters the possible physical interface address by comparing it to known popular VM Software adresses
+// and Teredo Tunneling Pseudo-Interface.
+func isPhysicalInterface(addr *net.HardwareAddr) bool {
+	var err *error = nil
+
+	// Mac Address parts to look for, and identify non physical devices. There may be more, update me!
+	var macAddrPartsToCheck []string = []string{
```

Comment:
This can be a const at the top of the file. Can you also rename it to `macAddrPartsToFilter`. Will make it clean when someone looks at the file what it is for.